### PR TITLE
Resource and operator managers

### PR DESCRIPTION
**Pull request by rkubis**

### CODE REVIEW

File: systemtests/README.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+System tests for Apicurio Registry and Apicurio Registry operators.
+
+## Necessary environment variables
+
+Necessary environment variables for run of system tests. Other environment variables are optional and have default values. See [Constants.java](/systemtests/src/main/java/io/apicurio/registry/systemtest/framework/Constants.java) for list of supported environment variables and their default values.
+
+|Environment variable|Purpose|
+|---|---|
+|`APICURIO_REGISTRY_OLM_OPERATOR_CATALOG_SOURCE_IMAGE`|Path/URL to image for CatalogSource of Apicurio Registry OLM operator|
```
Comment (obabec) [inline systemtests/README.md:11]: I would go with shorter env vars.

File: systemtests/README.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+|`APICURIO_REGISTRY_OLM_OPERATOR_SUBSCRIPTION_PACKAGE`|Package name for Subscription of Apicurio Registry OLM operator|
```
Comment (obabec) [inline systemtests/README.md:12]: Are you sure this is necessary?

File: systemtests/pom.xml [hunk 3]
```diff
@@ -72,6 +72,15 @@
...
             <version>2.2.0.Final</version>
         </dependency>
+        <dependency>
+            <groupId>org.junit.jupiter</groupId>
+            <artifactId>junit-jupiter-api</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>io.strimzi</groupId>
+            <artifactId>api</artifactId>
+            <version>0.28.0</version>
```
Comment (obabec) [inline systemtests/pom.xml:82]: Version should be a variable.

File: systemtests/src/main/java/io/apicurio/registry/systemtest/framework/Constants.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
+package io.apicurio.registry.systemtest.framework;
```
Comment (obabec) [inline systemtests/src/main/java/io/apicurio/registry/systemtest/framework/Constants.java:1]: All the constants are extremely long and extremely hard to use. I would rather see some Environment class that loads all the variables at the start of the test suite. And there could be ternary operators which could add the defaults.

Comment (obabec) [reply systemtests/src/main/java/io/apicurio/registry/systemtest/framework/Constants.java:1]: Something like this https://github.com/EnMasseProject/enmasse/blob/development/systemtests/src/main/java/io/enmasse/systemtest/Environment.java Where all envs are in the same place.

File: systemtests/src/main/java/io/apicurio/registry/systemtest/framework/Utils.java [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+import io.apicurio.registry.systemtest.time.TimeoutBudget;
+import io.fabric8.kubernetes.api.model.apps.StatefulSet;
+import io.fabric8.openshift.client.OpenShiftClient;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.slf4j.Logger;
+
+import java.nio.file.Paths;
+import java.time.Duration;
+
+public class Utils {
```
Comment (obabec) [inline systemtests/src/main/java/io/apicurio/registry/systemtest/framework/Utils.java:17]: I would choose a different name for this class.

File: systemtests/src/main/java/io/apicurio/registry/systemtest/framework/Utils.java [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+    private static final Logger utilsLogger = LoggerUtils.getLogger();
```
Comment (obabec) [inline systemtests/src/main/java/io/apicurio/registry/systemtest/framework/Utils.java:18]: logger is static value so naming should be something like UTILS_LOGGER. Overall I think it makes very little sense to change the name of the logger. Set loggers as LOGGER everywhere.

File: systemtests/src/main/java/io/apicurio/registry/systemtest/operator/types/OperatorKind.java [hunk 7]
```diff
@@ -0,0 +1,8 @@
+package io.apicurio.registry.systemtest.operator.types;
+
+public interface OperatorKind {
```
Comment (obabec) [inline systemtests/src/main/java/io/apicurio/registry/systemtest/operator/types/OperatorKind.java:3]: Wouldnt it be better to use enumeration?

**Review by obabec — COMMENTED** [review]
Overall it looks good, but there is still a lot to think about and work on.

**Review by EricWittmann — APPROVED** [review]

**Comment by rkubis** [comment]
API model is fixed, all fails are resolved, PR is ready for Squash and merge.

**Pull request merged**